settings.gradle

```diff
@@ -1,2 +1,5 @@
 rootProject.name = 'alfresco-hotfix-ALF-22094'
 
+include ':core'
```

Comment:
There are two options to handle a shared code base:

1. Put shared code in a `core` jar and make other subprojects depend on this jar
2. Put shared code in a shared directory and include them as `sourceSets` in the other subprojects

Now we have a combination of the two?
- a `core.jar` is build
- but we don't use it in the other subprojects and just include the sourceSets.

Reply:
Yes, I think I'm missing a trick. I will clarify

I did this so the dependencies are detected by IntelliJ and a default dependency is available while developing. 

The problem I had is that the jar needs to be compiled with different dependencies for different Alfresco versions.
So for option one, there should then be a differentiation in the jar task for example
for option two, development is hard as dependencies are not availble in java code in your IDE. Also running :amp on the root project will fail

But as said, I am probably just missing a tiny step that fixes all this

Reply:
As discussed: good for now, setup can be improved in later PR's.

Reply:
Please use (1), not (2).

Reply:
@tgeens 
Only problem I might see with (1) is that's is not just a simple standalone jar anymore. (Thinking about building as a SM)

Reply:
if that's the goal, shadowjar is the proper way to fix that.

Importing a shared-source-folder in multiple projects using `sourceSets` was a very painful mess we just git rid of in DE ?

Reply:
See also: APIX development in an IDE is misery

Reply:
Correct

Reply:
Please continue on #3 so this conversation is easy to find back when we address this issue